cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/JbodStorageTest.java

```diff
@@ -0,0 +1,282 @@
+            emptyMap(), emptyMap(), emptyMap()) { };
+
+    private Vertx vertx;
+    private Kafka kafka;
+    private KubernetesClient mockClient;
+    private KafkaAssemblyOperator kao;
+
+    private List<SingleVolumeStorage> volumes;
+
+    private void init() {
+
+        this.volumes = new ArrayList<>(2);
+
+        volumes.add(new PersistentClaimStorageBuilder()
+                .withId(0)
+                .withDeleteClaim(true)
+                .withSize("100Gi").build());
+        volumes.add(new PersistentClaimStorageBuilder()
+                .withId(1)
+                .withDeleteClaim(false)
+                .withSize("100Gi").build());
+
+        this.kafka = new KafkaBuilder()
+                .withNewMetadata()
+                    .withNamespace(NAMESPACE)
+                    .withName(NAME)
+                .endMetadata()
+                .withNewSpec()
+                    .withNewKafka()
+                        .withReplicas(3)
+                        .withNewJbodStorageStorage()
+                            .withVolumes(volumes)
+                        .endJbodStorageStorage()
+                    .endKafka()
+                    .withNewZookeeper()
+                        .withReplicas(1)
+                    .endZookeeper()
+                .endSpec()
+                .build();
+
+        // setting up the Kafka CRD
+        CustomResourceDefinition kafkaAssemblyCrd = Crds.kafka();
+
+        // setting up a mock Kubernetes client
+        this.mockClient = new MockKube()
+                .withCustomResourceDefinition(kafkaAssemblyCrd, Kafka.class, KafkaAssemblyList.class, DoneableKafka.class)
+                .end()
+                .build();
+
+        Crds.kafkaOperation(this.mockClient).inNamespace(NAMESPACE).withName(NAME).create(this.kafka);
+
+        this.vertx = Vertx.vertx();
+
+        // creating the Kafka operator
+        ResourceOperatorSupplier ros =
+                new ResourceOperatorSupplier(this.vertx, this.mockClient, false, 60_000L);
+
+        this.kao = new KafkaAssemblyOperator(this.vertx, false, 2_000, new MockCertManager(), ros, VERSIONS);
+    }
+
+    @Test
+    public void testCreatePersistentVolumeClaims(TestContext context) {
+
+        this.init();
+
+        Async async = context.async();
+        this.kao.reconcileAssembly(new Reconciliation("test-trigger", ResourceType.KAFKA, NAMESPACE, NAME), ar -> {
+            context.assertTrue(ar.succeeded());
+
+            List<PersistentVolumeClaim> pvcs = getPvcs(NAMESPACE, NAME);
+
+            for (int i = 0; i < this.kafka.getSpec().getKafka().getReplicas(); i++) {
+                int podId = i;
+                for (SingleVolumeStorage volume : this.volumes) {
+                    if (volume instanceof PersistentClaimStorage) {
+                        context.assertTrue(pvcs.stream().anyMatch(pvc -> {
+                            String pvcName = AbstractModel.VOLUME_NAME + "-" + volume.getId() + "-"
```

Comment:
Can we factor out into some common test function this concatenation of names to construct pvc names?

Reply:
There is already a method for that :-) Forgot to use it here.